source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRHostConfig.mm

```diff
@@ -0,0 +1,144 @@
+//
+//  ACRHostConfig.mm
+//  ACRHostConfig.h
+//
+//  Created by jwoo on 6/7/17.
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import "ACRHostConfig.h"
+
+using namespace AdaptiveCards;
+
+@implementation ACRHostConfig
+
+- (id) init {
+    self = [super init];
+
+    if(self)
+    { 
+        _m_hostConfig = std::make_shared<HostConfig>();
+    }
+
+    return self;
+}
+
+- (u_int32_t) retrieveIntFromHostConfigString: (NSString *) str{
+    u_int32_t num = 0;
+    NSScanner* scanner;
+    scanner = [NSScanner scannerWithString: str];
+    [scanner setScanLocation:1];
+    [scanner scanHexInt:&num];
+    return num;
+}
+
+- (UIColor* ) getTextBlockColor:(std::shared_ptr<TextBlock>) txtBlock{
+    u_int32_t num = 0;
+    std::string str;
+    switch (txtBlock->GetTextColor()) {
+        case TextColor::Dark: {	
+            str = self.m_hostConfig->colors.dark.normal;
+            break;
+        }
+        case TextColor::Light: {
+            str = self.m_hostConfig->colors.light.normal;
+            break;
+        }
+        case TextColor::Accent: {
+            str = self.m_hostConfig->colors.accent.normal;
+            break;
+        }
+        case TextColor::Good: {
+            str = self.m_hostConfig->colors.good.normal;
+            break;
+        }
+        case TextColor::Warning: {
+            str = self.m_hostConfig->colors.warning.normal;
+            break;
+        }
+        case TextColor::Attention: {
+            str = self.m_hostConfig->colors.attention.normal;
+            break;
+        }
+        default: {
+            str = self.m_hostConfig->colors.good.normal;
+            break;
+        }
+    }
+    
+    num = [self retrieveIntFromHostConfigString:[[NSString alloc] initWithCString:str.c_str()
+                                encoding:[NSString defaultCStringEncoding]]];
```

Comment:
It doesn't seem necessary to convert std::string -> NSString -> int. You can convert without going through NSString:

`std::stoul(str.substr(1), nullptr, 16)`